big-red-shuttle/IconView.swift

```diff
@@ -0,0 +1,184 @@
+        
+        return shapeLayer
+    }
+    
+    internal func drawTriangle() -> CAShapeLayer {
+        let desiredLineWidth:CGFloat = 5    // your desired value
+        
+        let trianglePath = UIBezierPath()
+        trianglePath.move(to: CGPoint(x: circleCenter - offset, y: circleViewHeight))
+        trianglePath.addLine(to: CGPoint(x: circleCenter + offset, y: circleViewHeight))
+        trianglePath.addLine(to: CGPoint(x: circleCenter, y: circleViewHeight + offset + triangleHeight))
+        trianglePath.close()
+        
+        let shapeLayer = CAShapeLayer()
+        shapeLayer.path = trianglePath.cgPath
+        
+        shapeLayer.fillColor = UIColor.iconblack.cgColor
+        shapeLayer.strokeColor = UIColor.iconblack.cgColor
+        shapeLayer.lineWidth = desiredLineWidth
+        
+        return shapeLayer
+    }
+    
+    internal func drawGroundBlackCircle(yPos:CGFloat) -> CAShapeLayer {
+        let desiredLineWidth:CGFloat = 1    // your desired value
+        
+        let circlePath = UIBezierPath(
+            arcCenter: CGPoint(x: CGFloat(circleCenter), y:yPos),
+            radius: CGFloat(5),
+            startAngle: CGFloat(0),
+            endAngle:CGFloat(M_PI * 2),
+            clockwise: true)
+        
+        let shapeLayer = CAShapeLayer()
+        shapeLayer.path = circlePath.cgPath
+        
+        shapeLayer.fillColor = UIColor.black.cgColor
+        shapeLayer.strokeColor = UIColor.black.cgColor
+        shapeLayer.lineWidth = desiredLineWidth
+        
+        return shapeLayer
+    }
+    
+    internal func drawGroundWhiteCircle(yPos:CGFloat) -> CAShapeLayer {
+        let desiredLineWidth:CGFloat = 1    // your desired value
+        
+        let circlePath = UIBezierPath(
+            arcCenter: CGPoint(x: CGFloat(circleCenter), y:yPos),
+            radius: CGFloat(8),
+            startAngle: CGFloat(0),
+            endAngle:CGFloat(M_PI * 2),
+            clockwise: true)
+        
+        let shapeLayer = CAShapeLayer()
+        shapeLayer.path = circlePath.cgPath
+        
+        shapeLayer.fillColor = UIColor.iconwhite.cgColor
+        shapeLayer.strokeColor = UIColor.iconwhite.cgColor
+        shapeLayer.lineWidth = desiredLineWidth
+        
+        return shapeLayer
+    }
+    
+    internal func drawGroundRedCircle(yPos:CGFloat) -> CAShapeLayer {
+        let desiredLineWidth:CGFloat = 1    // your desired value
+        
+        let circlePath = UIBezierPath(
+            arcCenter: CGPoint(x: CGFloat(circleCenter), y:yPos),
+            radius: CGFloat(13),
+            startAngle: CGFloat(0),
+            endAngle:CGFloat(M_PI * 2),
+            clockwise: true)
+        
+        let shapeLayer = CAShapeLayer()
+        shapeLayer.path = circlePath.cgPath
+        
+        shapeLayer.fillColor = UIColor.brsred.cgColor
+        shapeLayer.strokeColor = UIColor.brsred.cgColor
+        shapeLayer.lineWidth = desiredLineWidth
+        
+        return shapeLayer
```

Comment:
seems like the three draw circle functions have repetitive code, so let's refactor them to use a shared helper function